docs/src/theme/NavbarItem/DocsVersionDropdownNavbarItem.tsx

```diff
@@ -3,39 +3,145 @@
-import { useActiveDocContext } from "@docusaurus/plugin-content-docs/client";
-import type { WrapperProps } from "@docusaurus/types";
-import type DocsVersionDropdownNavbarItemType from "@theme/NavbarItem/DocsVersionDropdownNavbarItem";
-import DocsVersionDropdownNavbarItem from "@theme-original/NavbarItem/DocsVersionDropdownNavbarItem";
+import { translate } from "@docusaurus/Translate";
+import {
+	useVersions,
+	useActiveDocContext,
+	useDocsVersionCandidates,
+	useDocsPreferredVersion,
+} from "@docusaurus/plugin-content-docs/client";
+import type {
+	GlobalVersion,
+	GlobalDoc,
+	ActiveDocContext,
+} from "@docusaurus/plugin-content-docs/client";
+import { useLocation } from "@docusaurus/router";
+import type { LinkLikeNavbarItemProps } from "@theme/NavbarItem";
+import DefaultNavbarItem from "@theme/NavbarItem/DefaultNavbarItem";
+import type { Props } from "@theme/NavbarItem/DocsVersionDropdownNavbarItem";
+import DropdownNavbarItem from "@theme/NavbarItem/DropdownNavbarItem";
 import React from "react";
 
 /**
- * Note: this module was generated by `swizzling` the Docusaurus "classic" theme's Footer component.
- *
- * Its inner layout and contents were updated to better suite our needs, but the underlying styling and inner
- * components remain unchanged.
+ * Gets the documentation page marked as the main/landing page for a version,
+ * identified by the mainDocId property.
  *
- * See {@link https://docusaurus.io/docs/swizzling/ | here} for more information on swizzling.
+ * @param version - The version object to get the main doc from
+ * @returns The main documentation page for this version
  */
+function getVersionMainDoc(version: GlobalVersion): GlobalDoc {
+	// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+	return version.docs.find((doc) => doc.id === version.mainDocId)!;
+}
 
 /**
- * {@link DocsVersionDropdownNavbarItemWrapper} component props.
+ * When navigating between versions, attempts to keep the user on the same page.
+ * If the current page doesn't exist in the target version, falls back to that version's main page.
+ *
+ * @param version - The version being navigated to
+ * @param activeDocContext - Information about the currently viewed documentation
+ * @returns Either the equivalent page in the target version, or that version's main page
  */
-type Props = WrapperProps<typeof DocsVersionDropdownNavbarItemType>;
+function getVersionTargetDoc(
+	version: GlobalVersion,
+	activeDocContext: ActiveDocContext,
+): GlobalDoc {
+	return activeDocContext.alternateDocVersions[version.name] ?? getVersionMainDoc(version);
+}
+
+type AccessibleLinkProps = LinkLikeNavbarItemProps & {
+	"aria-label"?: string;
+	"role"?: string;
+	"aria-setsize"?: number;
+	"aria-posinset"?: number;
+	"className"?: string;
+};
 
 /**
- * Wraps the default DocsVersionDropdownNavbarItem to omit the drop-down on non-versioned pages.
+ * Note: this module was generated by `swizzling` the Docusaurus "classic" theme's Footer component.
```

Comment:
I think we can replace `Note:` by TSDoc's `@remarks`.

Reply:
updated